### PR TITLE
Dynamic Choice UI and Emote window

### LINKED ISSUE #8328: Add an emote menu for using emotes quicker.
### Description

Bind to do backflips, very important in.....combat....

Just a bind that you can press to do a backflip, pretty straightforward

### PR DESCRIPTION
**Pull request by MaxIsJoe**
https://user-images.githubusercontent.com/34368774/218246116-6ea08d81-b3e5-46c5-b654-229445e167a2.mp4

final look:
image

closes #8328

CL: [New] Added an emote window for those who keep messing up their `*me`s or have problem spelling words. This menu can be accessed by pressing your local chat keybind + shift and will display all available emotes that can be triggered from chat.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
this is done and ready for review

File: UnityProject/Assets/Scripts/Util/StringExt.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿namespace Util
+{
+	internal static class StringExt
+	{
+		/// <summary>
+		/// Capitalizes the first letter in a string. Does not check for multiple words, only the first character in the string array.
+		/// </summary>
+		/// <param name="str">The string you want to manipulate.</param>
+		/// <returns>The same string with the first character in upper case.</returns>
+		public static string CapitalizeFirstLetter(this string str)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Util/StringExt.cs]: I'm sure this already exists

File: UnityProject/Assets/Scripts/Core/Chat/EmoteActionManager.cs [hunk 2]
```diff
@@ -10,6 +12,45 @@ public class EmoteActionManager : SingletonManager<EmoteActionManager>
 		[SerializeField] private EmoteListSO emoteList;
 		public EmoteListSO EmoteList => emoteList;
 
+		private KeybindManager.KeyCombo windowInputComboKeys = new KeybindManager.KeyCombo();
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Chat/EmoteActionManager.cs]: you keep adding hardcoded keys for new features. Can't we use the input system we already have?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/EmoteActionManager.cs]: ???
The keybind manager checks for actions based on keybinds the users set

File: UnityProject/Assets/Scripts/Util/StringExt.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Util/StringExt.cs]: We have functions to capitalize the whole string, but not the first one.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Util/StringExt.cs]: I'm sure I added that extension, also one that will capitalize every first letter in a sentence.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Util/StringExt.cs]: Found it:
https://github.com/unitystation/unitystation/blob/3c52c81a360e411f89463f11236bb692041c5d3e/UnityProject/Assets/Scripts/Util/SweetExtensions.cs#L513

Feel free to move it if you wish, it is true it is very hard to find in that chaotical collection of extensions.

**Review by corp-0 — APPROVED** [review]

**Pull request merged**